cmd/habitat/community/manager.go

```diff
@@ -1,47 +1,92 @@
 package community
 
 import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"io/ioutil"
 	"os"
 	"path"
 	"path/filepath"
+	"sync"
+	"time"
 
 	"github.com/eagraf/habitat/cmd/habitat/community/consensus/cluster"
+	"github.com/eagraf/habitat/cmd/habitat/community/state"
+	"github.com/eagraf/habitat/cmd/habitat/procs"
 	"github.com/eagraf/habitat/cmd/habitat/proxy"
 	"github.com/eagraf/habitat/pkg/compass"
 	"github.com/eagraf/habitat/pkg/ipfs"
 	"github.com/eagraf/habitat/structs/community"
 	"github.com/google/uuid"
 	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/rs/zerolog/log"
 )
 
 type Manager struct {
-	Path    string
-	config  *ipfs.IPFSConfig
-	p2pHost host.Host
+	Path           string
+	config         *ipfs.IPFSConfig
+	p2pHost        host.Host
+	processManager procs.Manager
 
-	clusterManager *cluster.ClusterManager
-	communities    []*community.Community
+	clusterManager  *cluster.ClusterManager
+	communities     map[string]*state.CommunityStateMachine
+	communitiesLock *sync.Mutex
```

Comment:
why lock? are there code paths that can be called from different goroutines?

Reply:
Yeah, they get called from the community handlers that are endpoints for the server receiving stuff from `habitatctl`